pkgs/by-name/ty/typst/with-packages.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  buildEnv,
+  typstPackages,
+  makeBinaryWrapper,
+}:
+
+let
+  typstPackagesFor =
+    typst:
+    lib.makeOverridable (
+      { ... }@typstPkgs:
+      f:
+      let
+        paths = f typstPkgs;
+      in
+      buildEnv {
+        name = "${typst.name}-env";
+
+        inherit paths;
+
+        nativeBuildInputs = [ makeBinaryWrapper ];
+
+        postBuild = ''
+          export TYPST_LIB_DIR="$out/lib/typst/packages"
+          mkdir -p $TYPST_LIB_DIR/preview
+
+          for path in $(find $out -type l); do
+            mv $path $TYPST_LIB_DIR/preview
+          done
+
+          cp -r ${typst}/share $out/share
+          mkdir -p $out/bin
+
+          makeWrapper "${lib.getExe typst}" "$out/bin/typst" --set TYPST_PACKAGE_CACHE_PATH $TYPST_LIB_DIR
```

Comment:
This sounds like it might break ad-hoc package downloads https://github.com/typst/packages?tab=readme-ov-file#downloads

Reply:
Please see my comment above